code/client/src/components/WalletAddress.jsx

```diff
@@ -51,25 +52,25 @@ const WalletAddressOptions = ({ displayAddress, address, network, onAddressStyle
 }
 
 const MOUSE_HOVER_DETECTION_DELAY = 1000
+
 /**
  * Renders the provided wallet's address in either ONE style or normal.
  * Provides the ability to copy the address and link to wallet explorer.
  */
-const WalletAddress = ({ isMobile, wallet, network }) => {
-  const oneAddress = util.safeOneAddress(wallet.address)
+const WalletAddress = ({ isMobile, wallet, network, addressOverride, shortAddress }) => {
```

Comment:
It seems only two things from wallet are used here: address, name. And the name is solely used to determine whether an address should be shortened. Wouldn't it be cleaner if only `address` and `shorten` are arguments?

I don't think `network` is necessary either. You can use a selector inside this component  to get the current network

Reply:
Good point, I've refactored this component.